app/controllers/application_controller.rb

```diff
@@ -1,2 +1,34 @@
 class ApplicationController < ActionController::Base
+  private
+
+  def authenticate_user!
+    redirect_to root_path, alert: "You must be logged in to do that." unless user_signed_in?
+  end
+
+  def current_user
```

Comment:
I dont think this should be in private method if this is going to be used outside of this controller